web/src/main/java/org/wso2/testgrid/web/api/TestPlanService.java

```diff
@@ -286,6 +287,36 @@ public Response getTestSummary(@PathParam("id") String id) {
         }
     }
 
+    /**
+     * Returns the product performance dashboard url for the given test plan id
+     *
+     * @param id test plan id
+     * @return The requested roduct performance dashboard url for the test plan
+     */
+    @GET
+    @Path("/perf-url/{id}")
+    public Response getGrafanaURL(@PathParam("id") String id) {
+        try {
+
+            String dashURL = ConfigurationContext.getProperty(ConfigurationProperties.GRAFANA_URL) +
+                    "&from=now%2FM&to=now%2FM&var-VM=All&var-TestPlan=" + id;
+            if (StringUtil.isStringNullOrEmpty(dashURL)) {
+                String msg = "No test plan found for the given id " + id;
```

Comment:
Is this error message correct? I think we can't say "No test plan found for the given id" by just checking a string.